tensorflow_datasets/image/fgvc_aircraft.py

```diff
@@ -0,0 +1,151 @@
+            gen_kwargs={
+                "file_names": train_list,
+                "data_path": extract_path
+            },
+        ),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.VALIDATION,
+            gen_kwargs={
+                "file_names": val_list,
+                "data_path": extract_path
+            },
+        ),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TEST,
+            gen_kwargs={
+                "file_names": test_list,
+                "data_path": extract_path
+            },
+        ),
+    ]
+
+  def _generate_examples(self, file_names, data_path):
+    """Yields examples."""
+    def get_data(data_path, data_dict, train, trainval, test):
+      t_paths = [os.path.join(data_path, t_path)
+                 for t_path in [train, trainval, test]]
+      for t_path in t_paths:
+        with tf.io.gfile.GFile(t_path) as f:
+          t_data = [(line.split()[0],
+                     " ".join(line.split()[1:])) for line in f.readlines()]
+          for image, t_type in t_data:
+            data_dict[image] = t_type
+
+    variants = dict()
+    v_train = "fgvc-aircraft-2013b/data/images_variant_train.txt"
+    v_val = "fgvc-aircraft-2013b/data/images_variant_val.txt"
+    v_test = "fgvc-aircraft-2013b/data/images_variant_test.txt"
+    get_data(data_path, variants, v_train, v_val, v_test)
+
+    families = dict()
+    f_train = "fgvc-aircraft-2013b/data/images_family_train.txt"
+    f_val = "fgvc-aircraft-2013b/data/images_family_val.txt"
+    f_test = "fgvc-aircraft-2013b/data/images_family_test.txt"
+    get_data(data_path, families, f_train, f_val, f_test)
+
+    manufacturers = dict()
+    m_train = "fgvc-aircraft-2013b/data/images_manufacturer_train.txt"
+    m_val = "fgvc-aircraft-2013b/data/images_manufacturer_val.txt"
+    m_test = "fgvc-aircraft-2013b/data/images_manufacturer_test.txt"
+    get_data(data_path, manufacturers, m_train, m_val, m_test)
+
+    bboxes = dict()
+    bbox_path = os.path.join(data_path,
+                             "fgvc-aircraft-2013b/data/images_box.txt")
+
+    image_dir_path = "fgvc-aircraft-2013b/data/images"
+
+    def normalize_bbox(bbox_side, image_side):
+      return min(bbox_side / image_side, 1.0)
+
+    with tf.io.gfile.GFile(bbox_path) as f:
+      bbox_data = [line.split() for line in f.readlines()]
+      for file_name, xmin, ymin, xmax, ymax in bbox_data:
+        #width and height are not provided, find with pillow library
+        open_file = PIL.Image.open(os.path.join(data_path,
+                                                image_dir_path,
+                                                file_name+".jpg"))
+        width, height = open_file.size
+        bboxes[file_name] = (normalize_bbox(float(xmin), width),
+                             normalize_bbox(float(ymin), height),
+                             normalize_bbox(float(xmax), width),
+                             normalize_bbox(float(ymax), height))
+
```

Comment:
Everything before this point should be done in a different method, executed in _split_generators(), since it does not depend on the particular train/valid/test split.